crates/pop-cli/src/commands/up/rollup.rs

```diff
@@ -1,29 +1,40 @@
 // SPDX-License-Identifier: GPL-3.0
 
 use crate::{
-	build::spec::{BuildSpecCommand, CodePathBuf, StatePathBuf},
+	build::spec::{
+		BuildSpecCommand, ChainType, CodePathBuf, GenesisArtifacts, RelayChain, StatePathBuf,
+	},
 	call::chain::Call,
 	cli::traits::*,
 	common::{
 		chain::{configure, Chain},
 		wallet::submit_extrinsic,
 	},
-	style::style,
+	deployment_api::{DeployRequest, DeployResponse, DeploymentApi},
+	style::{format_step_prefix, format_url, style},
 };
-use anyhow::{anyhow, Result};
+use anyhow::Result;
 use clap::Args;
-use pop_common::parse_account;
+use cliclack::spinner;
+use pop_common::{parse_account, templates::Template, Profile};
 use pop_parachains::{
-	construct_proxy_extrinsic, find_dispatchable_by_name, Action, Payload, Reserved,
+	construct_proxy_extrinsic, find_dispatchable_by_name, Action, DeploymentProvider, Parachain,
+	Payload, Reserved, SupportedChains,
+};
+use std::{
+	env,
+	path::{Path, PathBuf},
+	str::FromStr,
 };
-use std::path::{Path, PathBuf};
+use strum::VariantArray;
 use url::Url;
 
 type Proxy = Option<String>;
 
 const DEFAULT_URL: &str = "wss://paseo.rpc.amforc.com/";
 const HELP_HEADER: &str = "Chain deployment options";
 const PLACEHOLDER_ADDRESS: &str = "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty";
+const POP_API_KEY: &str = "POP_API_KEY";
```

Comment:
I'd use a name specific to PDP instead of Pop.
Something like `PDP_API_KEY` would be better in the sense that the user might have already exported this key in their system, and is using it in various different services.

Also, the API key is not expedited by Pop in any way.

Reply:
I think it's a good idea. As you mentioned, it makes it clear that we don't manage the `API_KEY`. Renamed in https://github.com/r0gue-io/pop-cli/pull/459/commits/d1ec2d874f9a8093d2f8f9ed758354efc69bc0d0